code/modules/mob/living/silicon/robot/inventory.dm

```diff
@@ -1,82 +1,233 @@
-	. = FALSE
-	if(!(O in module.modules))
-		return
-	if(activated(O))
-		to_chat(src, "<span class='warning'>That module is already activated.</span>")
-		return
-	if(!held_items[1])
-		held_items[1] = O
-		O.screen_loc = inv1.screen_loc
-		. = TRUE
-	else if(!held_items[2])
-		held_items[2] = O
-		O.screen_loc = inv2.screen_loc
-		. = TRUE
-	else if(!held_items[3])
-		held_items[3] = O
-		O.screen_loc = inv3.screen_loc
-		. = TRUE
-	else
-		to_chat(src, "<span class='warning'>You need to disable a module first!</span>")
-	if(.)
-		O.equipped(src, ITEM_SLOT_HANDS)
-		O.mouse_opacity = initial(O.mouse_opacity)
-		O.layer = ABOVE_HUD_LAYER
-		O.plane = ABOVE_HUD_PLANE
-		observer_screen_update(O,TRUE)
-		O.forceMove(src)
-		if(istype(O, /obj/item/borg/sight))
-			var/obj/item/borg/sight/S = O
-			sight_mode |= S.sight_mode
-			update_sight()
+	if(held_items[module_num]) //If there's a held item, unequip it first.
+		if(!unequip_module_from_slot(held_items[module_num], module_num)) //If we fail to unequip it, then don't continue
+			return FALSE
+
+	switch(module_num)
+		if(1)
+			if(disabled_modules & BORG_MODULE_ALL_DISABLED)
+				return FALSE
+		
+			inv1.icon_state = "inv1 +b"
+			disabled_modules |= BORG_MODULE_ALL_DISABLED
 
+		if(2)
+			if(disabled_modules & BORG_MODULE_TWO_DISABLED)
+				return FALSE
+		
+			inv2.icon_state = "inv2 +b"
+			disabled_modules |= BORG_MODULE_TWO_DISABLED
+
+		if(3)
+			if(disabled_modules & BORG_MODULE_THREE_DISABLED)
+				return FALSE
+		
+			inv3.icon_state = "inv3 +b"
+			disabled_modules |= BORG_MODULE_THREE_DISABLED
+
+	playsound(loc, 'sound/machines/warning-buzzer.ogg', 50, TRUE, TRUE)
+	audible_message("<span class='warning'>[src] sounds an alarm! \"SYSTEM ERROR: Module [module_num] OFFLINE.\"</span>", self_message = null)
+	to_chat(src, "<span class='userdanger'>SYSTEM ERROR: Module [module_num] OFFLINE.</span>")
+
+	return TRUE
+	
+
+/**
+  * Breaks all of a cyborg's slots. 
+  */
+/mob/living/silicon/robot/proc/break_all_cyborg_slots()
+	for(var/S in 1 to 3)
+		break_cyborg_slot(S)
+
+/**
+  * Repairs the slot number, updating the icon.
+  *
+  * Arguments
+  * * module_num - the module number being repaired.
+  */
+/mob/living/silicon/robot/proc/repair_cyborg_slot(module_num)
+	if(is_invalid_module_number(module_num, TRUE))
+		return FALSE
 
+	switch(module_num)
+		if(1)
+			if(!(disabled_modules & BORG_MODULE_ALL_DISABLED))
+				return FALSE
+		
+			inv1.icon_state = "inv1"
+			disabled_modules &= ~BORG_MODULE_ALL_DISABLED
+		if(2)
+			if(!(disabled_modules & BORG_MODULE_TWO_DISABLED))
+				return FALSE
+		
+			inv2.icon_state = "inv2"
+			disabled_modules &= ~BORG_MODULE_TWO_DISABLED
+		if(3)
+			if(!(disabled_modules & BORG_MODULE_THREE_DISABLED))
+				return FALSE
+		
+			inv3.icon_state = "inv3"
+			disabled_modules &= ~BORG_MODULE_THREE_DISABLED
+
+	to_chat(src, "<span class='notice'>ERROR CLEARED: Module [module_num] back online.</span>")
+
+	return TRUE
```

Comment:
would it be feasible to kill off the hardcoded inv1/2/3 or are they too baked into borg existence

so much code duplication